### PR TITLE
Resolve various Compiler warnings

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: contrib/libev/ev.c [hunk 1]
```diff
@@ -2146,7 +2146,7 @@ typedef struct
   #include "ev_wrap.h"
 
   static struct ev_loop default_loop_struct;
-  EV_API_DECL struct ev_loop *ev_default_loop_ptr = 0; /* needs to be initialised to make it a definition despite extern */
```
Comment (vstakhov) [inline contrib/libev/ev.c:2149]: This is left for a reason. It is proven that making this var static might cause multiple instantiation of it in `rspamd` binary and `librspamd-server.so`. This usually takes place when IPO is turned on. Hence, I would advice to revert this change please.

**Comment by vstakhov** [comment]
Very cool, thank you very much! The only thing I cannot merge for now is that `extern` default loop in libev...

File: contrib/libev/ev.c (hunk 1 above)
Comment (cgzones) [reply contrib/libev/ev.c]: Dropped the related commit

Comment (vstakhov) [reply contrib/libev/ev.c]: Thank you very much!
